operations/configuration_management/Target_Ansible_Tasks_for_Image_Customization.md

```diff
@@ -3,24 +3,72 @@
 The Configuration Framework Service \(CFS\) enables Ansible playbooks to run against both running nodes \(node personalization\) and images prior to boot\(image customization\).
 See [Configuration Management Use Cases](Configuration_Management.md#use-cases) for more information about image customization and when it should be used.
 
-Ideally image customization playbooks should be separate from node personalization playbooks. This reduces the likelihood that Ansible content will run in both modes, and reduces the need for conditional checks in the playbooks.
+## Using the a host group
 
-For situations when both image customization and node personalization need to be part of the same playbook, CFS provides the `cray_cfs_image` variable to distinguish between
-node personalization and image customization. When this variable is set to `true`, it indicates that the CFS session is an image customization and the playbook is targeting an image.
+Ansible is most efficient at skipping tasks when the don't apply to any nodes in the current `hosts` target.  For this reason it's recommended that Ansible plays first setup a host group using the `add_host` module, and then use this to target plays for image customization rather than using the `when` conditional.
 
-## Using `cray_cfs_image`
+```yaml
+- name: setup cfs_image host group
+  gather_facts: false
+  hosts: all
+  tasks:
+    - name: Add image customization hosts to a the cfs_image host group.
+      add_host:
+        name: '{{ inventory_hostname }}'
+        groups: cfs_image
+      when: cray_cfs_image
+```
+
+For ease of use, future versions of CFS will setup this group automatically during the inventory creation.  This won't significantly improve performance over using `add_host` task, but it will allow this task to be removed for clarity.  Creating a `cfs_image` group will align the with this future feature in CFS and make the conversion simple.  The `add_host` task will be able to be removed without changing any other parts of the playbook.
```

Comment:
Reworded and ordered:
For ease of use, future versions of CFS will set up this `cfs_image` group automatically during the inventory creation.  In preparation for this, create a `cfs_image` group _now_ to make that conversion simple in the future.  In CFS' next release, this `add_host` task can simply be removed without changing any other parts of the playbook.